server/src/main/java/org/apache/druid/client/selector/HistoricalFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.druid.client.selector;
+
+import com.google.common.collect.ImmutableSet;
+import it.unimi.dsi.fastutil.ints.Int2ObjectRBTreeMap;
+import org.apache.druid.client.QueryableDruidServer;
+
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class HistoricalFilter implements Function<Int2ObjectRBTreeMap<Set<QueryableDruidServer>>, Int2ObjectRBTreeMap<Set<QueryableDruidServer>>>
```

Comment:
Make `HistoricalFilter` an interface with a method `getQueryableServers`.
The implementation would be the `DynamicConfigManager` class itself.

```java
public Set<QueryableDruidServer> getQueryableServers(Set<QueryableDruidServer> allServers, CloneQueryMode mode) {
  // read the coordinator dynamic config here and ignore the servers from allServers as applicable
}
```

Reply:
Changing the function to Set has some issues, as mentioned in https://github.com/apache/druid/pull/17899#discussion_r2039275089. 
Regarding changing `DynamicConfigManager` to an implementation of `HistoricalFilter`, this means that the cloneQueryMode enum would need to be added as an additional argument to each function in ServerSelector. Is this still the cleaner option?

Reply:
Oh, using a set is not a requirement, I just wrote it like that for simplicity. Same goes for the enum arg too.

In essence, we should just make `HistoricalFilter` an interface with some filtration method rather than just using a `Function<A, B>` and have the view class be an implementation of `HistoricalFilter`. It would be a cleaner design that is easier to follow and test.

Reply:
Makes sense! Made this change.